### ISSUE TITLE
[bug] Moon doesn't handle files being moved/deleted

### ISSUE THREAD

**Issue by mrobinsn** [moonrepo/moon#937]
It seems like moon will choke until the cache is cleared if you rename or delete any files that are referenced by a file group. 

Steps to reproduce issue:
1. Initialize a git repository
2. Initialize moon with a project in the root
3. Populate `moon.yml`:
    ```yaml
    $schema: https://moonrepo.dev/schemas/project.json
    fileGroups:
        myFiles:
        - '**/*.txt'
    tasks:
        list:
        command:
        - ls
        - '@files(myFiles)'
        inputs:
        - '@files(myFiles)'
        platform: system
    ```
4. Create some test files that match the file group
    ```bash
    touch file-{1..3}.txt
    ```
5. Run the target:
    ```bash
    ❯ moon run :list
    ▪▪▪▪ moon-rename-repro:list
    ./file-1.txt
    ./file-2.txt
    ./file-3.txt
    ▪▪▪▪ moon-rename-repro:list (6ms, 44cbecbc)

    Tasks: 1 completed
    Time: 16ms
    ```
6. Rename or remove one of those files:
    ```bash
    rm file-3.txt
    ``` 
7. Run the target (fails):
    ```bash
    ❯ moon run :list
    ▪▪▪▪ moon-rename-repro:list
    ./file-1.txt
    ./file-2.txt
    ls: ./file-3.txt: No such file or directory
    ▪▪▪▪ moon-rename-repro:list (5ms, b36a2c0c)
    Error:   × Process ls failed with a 1 exit code.
    ```
8. Run the target without cache (works):
    ```bash
    ❯ moon run :list --cache=off
    ▪▪▪▪ moon-rename-repro:list
    ./file-1.txt
    ./file-2.txt
    ▪▪▪▪ moon-rename-repro:list (5ms, ed766218)

    Tasks: 1 completed
    Time: 15ms
    ```

This is a simple example, but you can see how this would come up pretty often when refactoring a code base.

**Comment by milesj** [comment]
@mrobinsn I looked into this and I'm not sure what the best solution is at the moment.

What's happening is that `@files` gets expanded to the list of files when creating the task, and then the task is inserted into the project graph with the literal paths. The graph is then cached to `.moon/cache/states/projectGraph.json`, so subsequent runs are using the cache (which has the literal paths).

That's what's causing the error. The only solution right now is to not cache the project graph, but that'll be a big performance hit. I'll look into other ways to solve this.

**Comment by mrobinsn** [comment]
Hey @milesj any new thoughts on this issue?

**Comment by milesj** [comment]
It'll be solved with https://github.com/moonrepo/moon/pull/980 Aiming for next release.

**Comment by milesj** [comment]
Fixed in v1.11 https://moonrepo.dev/blog/moon-v1.11#unlocked-features
